rsocket-transport-netty/src/main/java/io/rsocket/transport/netty/server/TcpServerTransport.java

```diff
@@ -47,15 +54,21 @@ public static TcpServerTransport create(TcpServer server) {
 
     @Override
     public Mono<Closeable> start(ConnectionAcceptor acceptor) {
-        return server
-            .newHandler((in, out) -> {
-                in.context().addHandler("server-length-codec", new RSocketLengthCodec());
-                NettyDuplexConnection connection = new NettyDuplexConnection(in, out, in.context());
-                acceptor.apply(connection).subscribe();
-
-                return out.neverComplete();
-            })
-            .map(NettyContextClosable::new);
+        Mono<? extends NettyContext> serverMono = server
+                .newHandler((in, out) -> {
+                    in.context().addHandler("server-length-codec", new RSocketLengthCodec());
+                    NettyDuplexConnection connection = new NettyDuplexConnection(in, out, in.context());
+                    acceptor.apply(connection).subscribe();
+
+                    return out.neverComplete();
+                });
+
+        serverMono.map(s -> s.address()).subscribe(address);
```

Comment:
If we provide port value of "0", each time we call start(), it generates a new NettyContext, which has a different "address" - "port" value. So if the caller first calls start(..) and then asks for the "address" value, while another caller also calls start(..) and asks for the address value, then there will be a race condition. Other than that, when we call start(..) at the inside of a single thread, the address value of the initial call will be lost.

It seems we only need the "address" if we pass port value of "0", otherwise address can be created by merging "bindAddress" and "port"? If so, we might have a overload method which also provides the address value back? If caller really needs the address back, the caller can call this overloaded one. So we will not need any class member and it will still be ok for multithreading?